src/src/pages/capabilities/jsonmetadata/index.js

```diff
@@ -0,0 +1,203 @@
+import React, { useContext, useEffect, useState } from "react";
+import SelectedCapabilityContext from "../SelectedCapabilityContext";
+import AppContext from "../../../AppContext";
+import { Button, ButtonStack, Spinner } from "@dfds-ui/react-components";
+import Ajv from "ajv";
+import SyntaxHighlighter from "react-syntax-highlighter";
+import styles from "./jsonmetadata.module.css";
+import PageSection from "../../../components/PageSection";
+import MonacoEditor, { useMonaco } from "@monaco-editor/react";
+import { vs as syntaxStyle } from "react-syntax-highlighter/dist/esm/styles/hljs";
+export function JsonMetadataWithSchemaViewer() {
+  const monaco = useMonaco();
+  const { metadata, setCapabilityJsonMetadata } = useContext(
+    SelectedCapabilityContext,
+  );
+  const [isEditing, setIsEditing] = useState(false);
+  const [isDirty, setIsDirty] = useState(false);
+  const { selfServiceApiClient } = useContext(AppContext);
+  const [currentMetadataString, setCurrentMetadataString] = useState(metadata);
+  const [jsonString, setJsonString] = useState("");
+  const [schemaString, setSchemaString] = useState("");
+  const [validationError, setValidationError] = useState("");
+
+  useEffect(() => {
+    if (monaco && schemaString) {
+      const monacoSchema = {
+        fileMatch: ["*"],
+        schema: JSON.parse(schemaString),
+      };
+      monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
+        validate: true,
+        schemas: [monacoSchema],
+      });
+    }
+  }, [schemaString, monaco]);
+
+  useEffect(() => {
+    if (metadata) {
+      const prettyMetadata = prettifyJsonString(metadata);
+      setCurrentMetadataString(prettyMetadata);
+      setJsonString(prettyMetadata);
+    }
+  }, [metadata]);
+
+  useEffect(() => {
+    if (schemaString === "") {
+      selfServiceApiClient.getCapabilityJsonMetadataSchema().then((schema) => {
```

Comment:
Using `then` is not -- to my knowledge -- kosher, when we could use `useEffect` hooks.
At least I don't like mixing and matching eventuality-handlers, and since we are using `useEffect` everywhere else, I believe we should stick to that.

Reply:
changed to await async